omim2obo/parsers/omim_txt_parser.py

```diff
@@ -233,36 +256,111 @@ def parse_mim2gene(lines, filename='mim2gene.tsv', filename2='genemap2.tsv') ->
-    #     '4': 'A contiguous gene deletion or duplication syndrome, multiple genes are deleted or duplicated causing '
-    #          'the phenotype.',
-    # }
-    ret = {}
-    p = re.compile(r".*,\s+(\d+)\s\(\d\)")
+def parse_morbid_map(lines) -> (Dict, Dict):
+    """Parse morbid map file. Part of this inspired by:
+    https://github.com/monarch-initiative/monarch-ingest/blob/main/monarch_ingest/ingests/omim/gene_to_disease.py"""
+
+    # Aggregate data by gene MIM
+    gene_mim_data = {}
+    # phenotype_label_regex: groups: (1) phenotype label, (2) MIM number (optional), (3) phenotype mapping key (optional)
+    phenotype_label_regex = re.compile(r'(.*), (\d{6})\s*(?:\((\d+)\))?')
+    # phenotype_label_regex: groups: (1) phenotype label, (2) phenotype mapping key (optional)
+    phenotype_label_no_mim_regex = re.compile(r'(.*)\s+\((\d+)\)')
     for line in lines:
         if line.startswith('#'):
             continue
-        tokens = line.split('\t')
-        if not tokens or tokens == ['']:
+        fields: List[str] = line.split('\t')
+        if not fields or fields == ['']:
             continue
-        m = p.match(tokens[0])
-        if m:
-            phenotype_mim_number = m.group(1)
+
+        phenotype_label_and_metadata: str = fields[0]
+        gene_symbols: List[str] = fields[1].split(', ')
+        gene_mim_number: str = fields[2].strip()  # todo: eventually would like this to be `int`
+        cyto_location: str = fields[3].strip()
+
+        phenotype_label, phenotype_mim_number, association_key = '', '', ''
+        label_data_with_mim_num = phenotype_label_regex.match(phenotype_label_and_metadata)
+        label_data_no_mim_num = phenotype_label_no_mim_regex.match(phenotype_label_and_metadata)
+
+        if label_data_with_mim_num:
+            phenotype_label, phenotype_mim_number, association_key = label_data_with_mim_num.groups()
+        elif label_data_no_mim_num:
+            phenotype_label, association_key = label_data_no_mim_num.groups()
         else:
-            phenotype_mim_number = ''
-        gene_mim_number = tokens[2].strip()
-        cyto_location = tokens[3].strip()
-        ret[gene_mim_number] = [phenotype_mim_number, cyto_location]
-    return ret
+            print(f'Warning: Failed to parse phenotype label in morbidmap.txt row: {line}', file=sys.stderr)
+
+        if gene_mim_number not in gene_mim_data:
+            gene_mim_data[gene_mim_number] = []
+        # phenotype_identifier: By design, I would say that there should always be a numeric ID for this, such as a
+        # ...MIM number. In actuality, sometimes there is no MIM num, e.g.:
+        #   Phenotype	Gene Symbols	MIM Number	Cyto Location
+        #   {Autism susceptibility 14A} (2)	DEL16p11.2, C16DELp11.2, AUTS14A	611913	16p11.2
+        phenotype_identifier = phenotype_mim_number if phenotype_mim_number else phenotype_label
```

Comment:
@matentzn I _really_ don't like this. I/we didn't realize this before, but what is inside the "Phenotype" column of `morbidmap.txt` is not a phenotype, but a "gene::phenotype association", with additional information (e.g. the mapping key, as we've discovered). This gene::phenotype association" has a label (what is shown in the "Phenotype" column is its label). There is sometimes (I previously thought "always") a phenotype MIM number inside of the "gene::phenotype association label". But the "gene::phenotype associations" themselves have no identifiers, other than the label (e.g. `{Autism susceptibility 14A} (2)` in this case).

```py
        # phenotype_identifier: By design, I would say that there should always be a numeric ID for this, such as a
        # ...MIM number. In actuality, sometimes there is no MIM num, e.g.:
        #   Phenotype	Gene Symbols	MIM Number	Cyto Location
        #   {Autism susceptibility 14A} (2)	DEL16p11.2, C16DELp11.2, AUTS14A	611913	16p11.2
        phenotype_identifier = phenotype_mim_number if phenotype_mim_number else phenotype_label
```

You can find a report of all of these such cases in the recent file I committed at the root of the repo: `omimIssues.json`

Reply:
Marking this resolved as I made some issues where we'll carry on the discussion.